HeapSort/HeapSort.cpp

```diff
@@ -41,4 +41,61 @@ void swap(int* a, int* b)
 	int temp = *a;
 	*a = *b;
 	*b = temp;
+}
+
+
+
+//*****************Heap Sort Revision***************
+
+void BuildMaxHeap(int arr[], int len);
+void Maxheapify(int arr[], int index, int len);
+void sswap(int &a, int&b);
+
+__declspec(dllexport) void heapSortRev(int arr[], int arrLen)
+{
+	int len = arrLen - 1;
+	BuildMaxHeap(arr, len);
+	
+	while(len > 0)
+	{
+		sswap(arr[0], arr[len]);
+		--len;
+		Maxheapify(arr, 0, len);
+	}
+}
+
+void BuildMaxHeap(int arr[], int len)
+{
+	for (int i = len / 2; i >= 0; i--)
+	{
+		Maxheapify(arr, i, len);
+	}
+}
+
+void Maxheapify(int arr[], int index, int len)
+{
+	int lc = 2 * index + 1;
+	int rc = 2 * index + 2;
+	int max = index;
+	
+	if (lc <= len && arr[max] < arr[lc])
+		max = lc;
+
+	if (rc <= len && arr[max] < arr[rc])
+		max = rc;
+
+	if (arr[max] == arr[index])
+		return;
+	else
+	{
+		sswap(arr[max], arr[index]);
+		Maxheapify(arr, max, len);
+	}
+}
+
+void sswap(int& a, int& b)
+{
+	size_t temp = a;
```

Comment:
here, to swap the ints
do we need temp var as size_t ?

Reply:
Incorporated this review comment. Changed it to int.